calico/src/main/scala/calico/html/Modifier.scala

```diff
@@ -0,0 +1,144 @@
+    Resource.eval {
+      F.delay {
+        e.appendChild(dom.document.createTextNode(s))
+        ()
+      }
+    }
+
+  inline given forStringSignal[E <: fs2.dom.Node[F]]: Modifier[F, E, Signal[F, String]] =
+    _forStringSignal.asInstanceOf[Modifier[F, E, Signal[F, String]]]
+
+  private val _forStringSignal: Modifier[F, dom.Node, Signal[F, String]] = (s, e) =>
+    s.getAndUpdates.flatMap { (head, tail) =>
+      Resource
+        .eval(F.delay(e.appendChild(dom.document.createTextNode(head))))
+        .flatMap { n =>
+          tail.foreach(t => F.delay(n.textContent = t)).compile.drain.cedeBackground
+        }
+        .void
+    }
+
+  inline given forStringOptionSignal[E <: fs2.dom.Node[F]]
+      : Modifier[F, E, Signal[F, Option[String]]] =
+    _forStringOptionSignal.asInstanceOf[Modifier[F, E, Signal[F, Option[String]]]]
+
+  private val _forStringOptionSignal: Modifier[F, dom.Node, Signal[F, Option[String]]] =
+    _forStringSignal.contramap(_.map(_.getOrElse("")))
+
+  given forResource[E <: fs2.dom.Node[F], A](
+      using M: Modifier[F, E, A]): Modifier[F, E, Resource[F, A]] =
+    (a, e) => a.flatMap(M.modify(_, e))
+
+  given forFoldable[E <: fs2.dom.Node[F], G[_]: Foldable, A](
+      using M: Modifier[F, E, A]): Modifier[F, E, G[A]] =
+    (ga, e) => ga.foldMapM(M.modify(_, e)).void
+
+  inline given forNode[N <: fs2.dom.Node[F], N2 <: fs2.dom.Node[F]]
+      : Modifier[F, N, Resource[F, N2]] =
+    _forNode.asInstanceOf[Modifier[F, N, Resource[F, N2]]]
+
+  private val _forNode: Modifier[F, dom.Node, Resource[F, dom.Node]] = (n2, n) =>
+    n2.evalMap(n2 => F.delay(n.appendChild(n2)))
+
+  inline given forNodeSignal[N <: fs2.dom.Node[F], N2 <: fs2.dom.Node[F]]
+      : Modifier[F, N, Signal[F, Resource[F, N2]]] =
+    _forNodeSignal.asInstanceOf[Modifier[F, N, Signal[F, Resource[F, N2]]]]
+
+  private val _forNodeSignal: Modifier[F, dom.Node, Signal[F, Resource[F, dom.Node]]] =
+    (n2s, n) =>
+      n2s.getAndUpdates.flatMap { (head, tail) =>
+        DomHotswap(head).flatMap { (hs, n2) =>
+          F.delay(n.appendChild(n2)).toResource *>
+            tail
+              .foreach(hs.swap(_)((n2, n3) => F.delay(n.replaceChild(n3, n2))))
+              .compile
+              .drain
+              .cedeBackground
+        }.void
+      }
+
+  inline given forNodeOptionSignal[N <: fs2.dom.Node[F], N2 <: fs2.dom.Node[F]]
+      : Modifier[F, N, Signal[F, Option[Resource[F, N2]]]] =
+    _forNodeOptionSignal.asInstanceOf[Modifier[F, N, Signal[F, Option[Resource[F, N2]]]]]
+
+  private val _forNodeOptionSignal
+      : Modifier[F, dom.Node, Signal[F, Option[Resource[F, dom.Node]]]] = (n2s, n) =>
+    Resource.eval(F.delay(Resource.pure[F, dom.Node](dom.document.createComment("")))).flatMap {
+      sentinel => _forNodeSignal.modify(n2s.map(_.getOrElse(sentinel)), n)
+    }
+
+  given forStringStream[E <: fs2.dom.Node[F]]: Modifier[F, E, Stream[F, String]] with
```

Comment:
Did you add these? Calico 0.2 will no longer allow `Stream`-based modifiers, only `Signal`.

Reply:
@armanbilge I guess I added these to make Example.scala compile. Now I've seen that you basically have removed Example.scala. I will revert this but I'm curious why you don't want to support `Stream` modifiers anymore? Obviously `Signal` makes more sense since it asserts that there is a value available from the start. But one potential shortcoming I can see is that there is no `Monad` instance for `Signal`, which makes it not possible to modify the value before the actual DOM elem is modified using another effect like a HTTP request for example. For this use case one has to introduce another `Signal` which gets modified in the background by getting a `Stream` from the initial `Signal`, which seems a bit involved.

Reply:
> potential shortcoming I can see is that there is no Monad instance for Signal

https://github.com/armanbilge/calico/blob/d0812e8009ea7b390292a0aee0279e23c9d9c302/frp/src/main/scala/calico/frp/frp.scala#L50

Reply:
Shame on me that I didn't know about this. Do you plan to push this upstream?

Reply:
Removed the Stream instances (again). What about Example.scala? I just modified it to use the `Signal` modifiers instead of `Stream` ones.

Reply:
Yes, I've discussed this with Fabio (FS2 maintainer), basically the idea is to "incubate" this instance in Calico, and if it proves to be robust then we can upstream it to FS2.

The implementation and/or test might still be a bit buggy, see:
- https://github.com/armanbilge/calico/issues/101

Reply:
It just went red after removing the instances

Reply:
Yeah, that test is flaking sometimes 😕 so far I have a hard time reproducing it. If you have any insight to that, it would be really great 😁

Reply:
Btw a couple more thoughts:

> Now I've seen that you basically have removed Example.scala

Yeah, actually I just use that as a sandbox for testing stuff, probably should rename it. All the examples go in the docs now.

> Obviously `Signal` makes more sense since it asserts that there is a value available from the start.

Yes, I wrote about this in the v0.1.2 release notes. Also it means we can make an optimization: you can retrieve the first value, render the HTML, and then setup the subscriptions in the background. But I still don't have this working perfectly.
https://github.com/armanbilge/calico/releases/tag/v0.1.2

Reply:
I'll try to reproduce locally